docs_src/design/adr/0020-UnitOfMeasure.md

```diff
@@ -0,0 +1,93 @@
+  units_standard: "UCUM"
+```
+   
+The unit of measure standard would be specified as a string in the device profile.  Note each device resource could specify a different unit of measure standard.  
+
+### Validation
+
+Initially, EdgeX would not validate the existence of the specification or standard used to specify the unit of measures.  Nor would EdgeX validate that the unit of measure value is a valid unit of measure per that specification.  The unit of measure specification is provided as additional (and optional) information which can be utilized in a way seen fit by the adopter.
+
+EdgeX may, in the future, provide a means to trigger validation of the unit of measure specification and of the unit of measures per that specification.  Doing so would require that EdgeX also have information about where to validate the specification's existence and valid units of measure (presumably a REST URL or other such confirmation site).
+
+In theory, the unit of measure could be validated on the "indbound" collection of the data (likely in a device service), on the "outbound" export of the data from EdgeX (likely in an application service), or in multiple locations if warranted.
+
+## Considersations
+
+It has been suggested that EdgeX borrow from the Open [Geospatial Consortium SensorThings](https://www.ogc.org/standards/sensorthings) standard and include a JSON object to provide more information about the UoM for the field.  An optional  unitOfMeasurement field would replace the optional units_standard in above (allowing for either standard or non-standard definitions but with more information).
+
+A JSON Object containing three key-value pairs. The name property presents the full name of the unitOfMeasurement; the symbol property shows the textual form of the unit symbol; and the definition contains the URI defining the unitOfMeasurement.
+
+From Example 4: A Datastream entity example:
+
+"unitOfMeasurement": {
+    "name": "degree Celsius",
+    "symbol": "°C",
+    "definition": "http://unitsofmeasure.org/ucum.html#para-30"
+}
+
+See https://docs.ogc.org/is/18-088/18-088.html#datastream
+
+## Consequences
+- Any validation could impact performance.
+- As with most validation, it should be configured to turn on or off by use case circumstances or trust of data providers.
+- Some unit of measures may need to be linked to some sort of group affilation or geo local.  For example, if an EdgeX instance was operating in the US, someone might want all distance measurements done in feet/inches, temperatures in Farenheit, and volume in gallons.  The concept of group or affiliation will be not part of the initial efforts to satisfy UoM association.
+
+## References
+
+### UoM Standards
+
+- https://ediacademy.com/blog/x12-unit-of-measurement-codes/
+- https://unece.org/fileadmin/DAM/cefact/recommendations/rec20/rec20_rev3_Annex2e.pdf
+- https://en.wikipedia.org/wiki/Unified_Code_for_Units_of_Measure
+- https://www.ogc.org/standards/sensorthings
```

Comment:
Per @farshidtz's suggestion, we should probably add a link to the sensor measurement lists (senML) RFC from the IETF: https://www.rfc-editor.org/rfc/rfc8428.html. This RFC actually presents a compact encoding scheme for device/sensor readings, and as such has a fair amount of overlap with the EdgeX schema. Since it was designed from the start to work on constrained networks, the JSON encoding uses very short element names (e.g. "u" vs. "Unit"). It also should be noted that in general, it seems to be a much more conservative set of units than some of the other standard (i.e. there's only one length units, "meter"; no support for "mile" or "light-year"), although this has been somewhat loosened by [RFC8798](https://www.rfc-editor.org/rfc/rfc8798.html) which proposes a new set of secondary units.

Reply:
added

Reply:
I also researched SenML and added a section addressing it.  Summary:  SenML might be something we want to look at for our base reading model, but doesn't really speak to unit of measure.